src/main/java/application/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,22 @@
+package application.dao;
+
+import application.model.Product;
+import application.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
+public interface ShoppingCartDao {
```

Comment:
methods:
- addProduct
- deleteProduct
should be at the service layer

Reply:
done